### DISCUSSION TITLE
How to config additional paths for Babel to compile

### DISCUSSION THREAD

**Discussion by yankustefan** [vercel/next.js#13532] [Help] [answered]
Hi,

I have a custom component library in the node_modules folder (or in .yalc), which I would like to compile together with the nextjs app.

I am using [Yalc](https://github.com/whitecolor/yalc), to work with yarn/npm packages locally.
Before, I was using create react app, and was able to accomplish that with [customize-cra](https://github.com/arackaf/customize-cra):

```
module.exports = function (config, env) {
  return Object.assign(config, override(
    disableEsLint(),
    addDecoratorsLegacy(),
    /*Make sure Babel compiles the stuff in these foldersr*/
    babelInclude([
      path.resolve('src'), // don't forget this
      path.resolve('.yalc/my-module/src'),
      path.resolve('node_modules/my-module/src')
    ])
    )(config, env)
  )
}
```

Without configuration it appears that the libraries' components (located in .yalc/my-module/src) can't find imported components (such as React, or a sibling component in the same library).

Now I would need to do a custom-webpack-config in [next.config.js](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config) ?

Do I need to modify defaultLoaders.babel directly or add a plugin ?

I would appreciate any input on how to accomplish that.

**Comment by Timer** [comment, marked as answer]
[`next-transpile-modules`](https://github.com/martpie/next-transpile-modules) is the go-to module for adding this behavior:

```js
// next.config.js
const withTM = require("next-transpile-modules")(["my-module", "my-module-2"]);

module.exports = withTM();
```

**Reply by yankustefan** [reply]
Thank you very much for your help!

**Reply by yankustefan** [reply]
Wonderful, it works like a charm!
